src/Validation/URLValidationRESTController.php

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * REST endpoint providing theme scan results.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AMP_Validated_URL_Post_Type;
+use AMP_Validation_Error_Taxonomy;
+use AmpProject\AmpWP\DevTools\UserAccess;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * URLValidationRESTController class.
+ *
+ * @since 2.1
+ */
+final class URLValidationRESTController extends WP_REST_Controller implements Delayed, Service, Registerable {
+
+	/**
+	 * URLValidationProvider instance.
+	 *
+	 * @var URLValidationProvider
+	 */
+	private $url_validation_provider;
+
+	/**
+	 * UserAccess instance.
+	 *
+	 * @var UserAccess
+	 */
+	private $dev_tools_user_access;
+
+	/**
+	 * Response schema.
+	 *
+	 * @var array
+	 */
+	protected $schema;
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @param URLValidationProvider $url_validation_provider URLValidationProvider instance.
+	 * @param UserAccess            $dev_tools_user_access UserAccess instance.
+	 */
+	public function __construct( URLValidationProvider $url_validation_provider, UserAccess $dev_tools_user_access ) {
+		$this->namespace               = 'amp/v1';
+		$this->url_validation_provider = $url_validation_provider;
+		$this->dev_tools_user_access   = $dev_tools_user_access;
+	}
+
+	/**
+	 * Registers all routes for the controller.
+	 */
+	public function register() {
+		register_rest_route(
+			$this->namespace,
+			'/validate-post-url/(?P<id>[\d]+)',
+			[
+				'args'   => [
+					'id'            => [
+						'description' => __( 'Unique identifier for the object.', 'amp' ),
+						'required'    => true,
+						'type'        => 'integer',
+					],
+					'preview_id'    => [
```

Comment:
Is the `id` ever different from the `preview_id` when the `preview_id` is supplied? It seems not. We can apparently skip passing it to the REST API endpoint.

Reply:
Good catch! The `preview_id` has been removed in 6fe44f9.